File: backend/bibliotheque/src/main/java/com/upb/bibliotheque/entity/Livre.java
package com.upb.bibliotheque.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.List;

@Entity
@Table(name = "livres")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Livre {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long livreId;

    @Column(nullable = false)
    private String titre;
    @Column(nullable = false)
    private String auteur;
    private String categorie;
    private String isbn;
    @Column(columnDefinition = "TEXT")
    private String description;
    private Integer exemplairesTotaux = 1;
    private Integer exemplairesDisponibles = 1;
    private String imageUrl;
    private String couleur3d;
    private Integer epaisseur3d;
    private Integer hauteur3d;

    @ManyToOne
    @JoinColumn(name = "catalogue_id")
    private Catalogue catalogue;

    @OneToMany(mappedBy = "livre", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Emprunt> emprunts;

    @OneToMany(mappedBy = "livre", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Reservation> reservations;

    @OneToMany(mappedBy = "livre", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Annotation> annotations;

    public boolean estDisponible() { return exemplairesDisponibles > 0; }
}

File: shared/models/livre.dart
class Livre {
  final int? livreId;
  final String titre;
  final String auteur;
  final String? categorie;
  final String? isbn;
  final String? description;
  final int exemplairesTotaux;
  final int exemplairesDisponibles;
  final String? imageUrl;
  final String? couleur3d;
  final int? epaisseur3d;
  final int? hauteur3d;

  Livre({
    this.livreId,
    required this.titre,
    required this.auteur,
    this.categorie,
    this.isbn,
    this.description,
    this.exemplairesTotaux = 1,
    this.exemplairesDisponibles = 1,
    this.imageUrl,
    this.couleur3d,
    this.epaisseur3d,
    this.hauteur3d,
  });

  factory Livre.fromJson(Map<String, dynamic> json) => Livre(
    livreId: json['livreId'],
    titre: json['titre'] ?? '',
    auteur: json['auteur'] ?? '',
    categorie: json['categorie'],
    isbn: json['isbn'],
    description: json['description'],
    exemplairesTotaux: json['exemplairesTotaux'] ?? 1,
    exemplairesDisponibles: json['exemplairesDisponibles'] ?? 1,
    imageUrl: json['imageUrl'],
    couleur3d: json['couleur3d'],
    epaisseur3d: json['epaisseur3d'],
    hauteur3d: json['hauteur3d'],
  );

  bool get estDisponible => exemplairesDisponibles > 0;
}